### PR TITLE
progress bar add write interface

### PR DESCRIPTION
**Pull request by zhijian-pro**
Based on [mbp](github.com/vbauerster/mpb) v7.0.3 version code modification, add write interface to it, to achieve the effect that the progress bar and the log will not overlap each other

### CODE REVIEW

**Comment by zhijian-pro** [comment]
**GC** 
image
**unit test**
image

**Review by davies — APPROVED** [review]

File: pkg/utils/progress.go [hunk 1]
```diff
@@ -150,6 +153,7 @@ func (p *Progress) Done() {
 		}
 	}
 	p.Progress.Wait()
+	SetOutput(os.Stderr, true)
```
Comment (SandyXSD) [inline pkg/utils/progress.go:156]: It's possible to use progress bar in a mounted client, e.g: backup will call DumpMeta

**Review by SandyXSD — CHANGES_REQUESTED** [review]
need fix

File: pkg/utils/logger.go [hunk 2]
```diff
@@ -129,3 +130,10 @@ func SetOutFile(name string) {
 		logger.tty = false
 	}
 }
+
+func SetOutput(w io.Writer, istty bool) {
+	for _, logger := range loggers {
+		logger.SetOutput(w)
+		logger.tty = istty
```
Comment (davies) [inline pkg/utils/logger.go:137]: tty is used by DisableLogColor(), we need to add `colorful` for that rather than using `tty=false`

**Comment by zhijian-pro** [comment]
> 
stderr stdout is redirected to a file during mount
image

image
Test backup meta when mount is ok
image

File: pkg/utils/logger.go [hunk 3]
```diff
@@ -126,6 +127,14 @@ func SetOutFile(name string) {
 	}
 	for _, logger := range loggers {
 		logger.SetOutput(file)
-		logger.tty = false
+		logger.colorful = false
+	}
+}
+
+func SetOutput(w io.Writer) {
+	colorful := isatty.IsTerminal(os.Stderr.Fd())
```
Comment (davies) [inline pkg/utils/logger.go:135]: We should not change it.

**Review by davies — APPROVED** [review]

**Review by SandyXSD — APPROVED** [review]

**Pull request merged**